packages/aws-cdk-lib/aws-cloudwatch/lib/graph.ts

```diff
@@ -651,7 +665,46 @@ export interface HorizontalAnnotation {
 }
 
 /**
- * Fill shading options that will be used with an annotation
+ * Vertical annotation to be added to a graph
+ */
+export interface VerticalAnnotation {
+  /**
+   * The date and time in the graph where the vertical annotation line is to appear
+   */
+  readonly date: Date;
```

Comment:
Honestly, I don't trust the `Date` type, and it's annoying to use. If you want to keep it, I'd recommend having something like:

```ts
export interface VerticalAnnotation {
  readonly dateTime?: Date;
  readonly dateTimeString?: string;
}
```

And then using runtime validation make sure exactly one of those is given (otherwise it's an error) and picking the one which is supplied.

Otherwise, I think I'd be okay just accepting the string variant.

Reply:
I am fine with not using the `Date` type. However, just using a normal string we would not have any validation for the value. Is that alright?

Reply:
We can validate it using a regex.

Reply:
Do you have any good at hand or is there any in the project? Otherwise I would use the 1 from https://www.regextester.com/97766 but I do not know how reliable it is.

Reply:
I took the 1 from https://www.regextester.com/97766. Please let me know if this needs adjustment.